## Boston Housing Prices

In this quiz we will predict the value of houses using 13 features in the Boston Housing prices dataset.

In [24]:
from sklearn.linear_model import LinearRegression
from sklearn.datasets import load_boston
import pandas as pd
import numpy as np

In [20]:
#Load data from sklearn sample datasets
boston_data = load_boston()

In [12]:
print(boston_data['DESCR'])

Boston House Prices dataset

Notes
------
Data Set Characteristics:  

    :Number of Instances: 506 

    :Number of Attributes: 13 numeric/categorical predictive
    
    :Median Value (attribute 14) is usually the target

    :Attribute Information (in order):
        - CRIM     per capita crime rate by town
        - ZN       proportion of residential land zoned for lots over 25,000 sq.ft.
        - INDUS    proportion of non-retail business acres per town
        - CHAS     Charles River dummy variable (= 1 if tract bounds river; 0 otherwise)
        - NOX      nitric oxides concentration (parts per 10 million)
        - RM       average number of rooms per dwelling
        - AGE      proportion of owner-occupied units built prior to 1940
        - DIS      weighted distances to five Boston employment centres
        - RAD      index of accessibility to radial highways
        - TAX      full-value property-tax rate per $10,000
        - PTRATIO  pupil-teacher ratio by town
      

In [13]:
x = boston_data['data']
y = boston_data['target']

In [18]:
print(x.shape)


(506, 13)


In [19]:
print(y.shape)

(506,)


We see that the X, the predictor matrix has 506 rows and 13 columns, one column for every feature
y, the outcome is an array of length 506 - and that's the median house price for the combination of features in every row.

In [21]:
# Make and fit the linear regression model
# TODO: Fit the model and assign it to the model variable
model = LinearRegression()
model.fit(x,y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=1, normalize=False)

In [26]:
array1 =  np.array([1,2,3])
print(array1)
array2 = np.array([[1,2,3]])
print(array2)
print(array1.shape)
print(array2.shape)

[1 2 3]
[[1 2 3]]
(3,)
(1, 3)


the second array is a row


In [28]:
# Make a prediction using the model - for the following x values as a row
sample_house = [[2.29690000e-01, 0.00000000e+00, 1.05900000e+01, 0.00000000e+00, 4.89000000e-01,
                6.32600000e+00, 5.25000000e+01, 4.35490000e+00, 4.00000000e+00, 2.77000000e+02,
                1.86000000e+01, 3.94870000e+02, 1.09700000e+01]]

prediction = model.predict(sample_house)

In [29]:
prediction

array([23.68420569])